script.js

```diff
@@ -1,4 +1,94 @@
+var getRandomHand = function () {
+  var randomDecimal = Math.random() * 3;
+  var randomInteger = Math.floor(randomDecimal); 
+   if (randomInteger == 0) {
+    return 'scissors';
+  }
+   if (randomInteger == 1) {
+    return 'paper';
+  }
+    return 'stone'; 
+  }; 
+ 
+var numWin = 0; 
+var numGames = 0; 
+var username = ''; 
+var mode = '';
+
+// main function 
 var main = function (input) {
-  var myOutputValue = 'hello world';
-  return myOutputValue;
-};
+
+// player name 
+if(!username){
+  if(!input ) {
+    return 'What is your your preferred name for the game?'; 
+  }
+    username = input; 
+    return `Hello ${username}! Hit the button again! `
+}
+
+// chooseing a mode 
+if(!mode){
+  if(!input ) {
+    return 'Which mode do you want to play? A. reverse B. Korean C. Comp vs Comp D. normal'; 
+  }
+    mode = input; 
+    var selection = function (){
+      if(input == 'A') {
+        return 'reverse'
+      }
+      if(input == 'B') {
+        return 'Korean'
+      }
+      if(input == 'C') {
+        return 'Comp vs Comp'
+      }
+      if(input == 'D') {
+        return 'normal'
+      }
+    };
+    var modename = selection(); 
+    return `You've chosen the ${modename} mode. <br><br>
+    Start the game by keying in any of the three words: scissors, paper, or stone.`; 
+}
+
+
+// userinput 
+  var playerinput = input; 
+ 
+  if(playerinput.substring(0,2) != 'sc' && playerinput.substring(0,2) != 'si' && playerinput.substring(0,2) != 'st' && playerinput.substring(0,2) != 'pa')
```

Comment:
This is a complicated piece of logic and needs a comment above it to explain. Please be liberal about comments- leave comments everywhere where they may be potential for a reader to not understand the code immediately.